src/Reflection/ReflectionClass.php

```diff
@@ -633,74 +633,38 @@ public function hasMethod(string $methodName): bool
 
     /**
      * Get an associative array of only the constants for this specific class (i.e. do not search
-     * up parent classes etc.), with keys as constant names and values as constant values.
-     *
-     * @return array<string, mixed>
-     */
-    public function getImmediateConstants(): array
-    {
-        return array_map(static fn (ReflectionClassConstant $classConstant): mixed => $classConstant->getValue(), $this->getImmediateReflectionConstants());
-    }
-
-    /**
-     * Get an associative array of the defined constants in this class,
-     * with keys as constant names and values as constant values.
-     *
-     * @return array<string, mixed>
-     */
-    public function getConstants(): array
-    {
-        return array_map(static fn (ReflectionClassConstant $classConstant): mixed => $classConstant->getValue(), $this->getReflectionConstants());
-    }
-
-    /**
-     * Get the value of the specified class constant.
-     *
-     * Returns null if not specified.
+     * up parent classes etc.), with keys as constant names and values as {@see ReflectionClassConstant} objects.
      *
-     * @return scalar|array<scalar>|null
+     * @return array<string, ReflectionClassConstant> indexed by name
      */
-    public function getConstant(string $name): string|int|float|bool|array|null
+    public function getImmediateConstants(int|null $filter = null): array
```

Comment:
If we only stick to `int $filter`, we should document its bitmask type precisely

Reply:
@Ocramius Can I prepare everything in next PR? I will change it for methods and properties as well.

Reply:
Sure 👍